main/observing/observation_run.py

```diff
@@ -630,3 +673,58 @@ def _shutdown_procedure(self, calibration, cooler=True):
             self.take_calibration_images()
         if cooler:
             self.camera.onThread(self.camera.cooler_set, False)
+
+    def threadcheck(self):
+        '''
+        Description
+        ----------
+        Checks to see if self.monitor has raised a crashed thread,
+        Restarts the crashed threads if there are any
+        
+        Returns
+        -------
+        None
+        '''
+        #print('---In thread Check')
+        threadlist = self.monitor.crash_return()
+        if self.monitor.threadcrash == True:
```

Comment:
threadcrash is a threading event, not a boolean, so I don't think this will work; threading events are mainly used for wait() statements to synchronize between threads, not send true/false information, but you could instead do something like "if self.monitor.threadcrash.isSet()"